src/Runtime/OMTensor.inc

```diff
@@ -276,8 +276,11 @@ OMTensor *omTensorCreateEmpty(
     return NULL;
 
   void *dataPtr = malloc(omTensorGetNumElems(tensor) * getDataTypeSize(dtype));
-  if (!dataPtr)
+  if (!dataPtr) {
+    // It is safe to free a null pointer (_allocatedPtr is null).
```

Comment:
The comment is out of context because you are not freeing a null ptr here, just calling `omTensorDestroy` (which might free `_allocatedPtr` even if NULL.  I would remove the comment.

Reply:
More complicated, the OMTensor is created with ownership, which means that it will try to deallocate `tensor->_allocatedPtr` which is set to null. I can remove the comment if that is too much info

Reply:
I guest what you really want to say here is to free `shape` and `strides` which are already allocated for the tensor, so that we can avoid memory leak when `malloc` fails.
 
If so, we could change the order of statements a bit to avoid this situation, e.g. do malloc for `dataPtr` first and then create a tensor. Something like:
```C
  int64_t numElem = 1;
  for (int64_t i = 0; i < rank; i++)
    numElem *= shape[i];

  void *dataPtr = malloc(numElem * getDataTypeSize(dtype));
  if (!dataPtr)
    return NULL;

  OMTensor *tensor = omTensorCreateWithOwnership(dataPtr, shape, rank, dtype, /*owning=*/true);
  if (!tensor) {
    free(dataPtr);
    return NULL;
  }

  return tensor;
```

Reply:
More concisely,
```
void *dataPtr;
OMTensor *tensor;
if ((dataPtr = malloc(getNumElems(shape, rank) * getDataTypeSize(dtype))) == NULL ||
    (tensor = omTensorCreateWithOwnership(dataPtr, shape, rank, dtype, /*owning=*/true)) == NULL) {
    free(dataPtr); // OK if dataPtr == NULL
    return NULL;
}
```

Reply:
Elegant, will replace with your suggestion.